Funcky/Internal/ValueMapper.cs

```diff
@@ -0,0 +1,12 @@
+namespace Funcky.Internal;
+
+internal static class ValueMapper
+{
+    public const int NotFoundValue = -1;
+
+    [Pure]
+    public static Option<int> MapNotFoundToNone(int index)
```

Comment:
Should we expose this as a public constructor of Option? Maybe something like `Option.FromIndex`?

Reply:
I think that should be another PR.